python/src/aiconfig/editor/server/server.py

```diff
@@ -177,8 +180,14 @@ async def run() -> FlaskResponse:
     state = get_server_state(app)
     aiconfig = state.aiconfig
     request_json = request.get_json()
+    cancellation_token_id: Union[str, None] = None
+    aiconfig_deep_copy: Union[AIConfigRuntime, None] = None
```

Comment:
I'm a little worried this is going to make the state even harder to reason about. We already have the file, the in-memory copy, and the frontend copy. This becomes an even bigger concern if we're allowing concurrent mutation requests (are we doing that here as a consequence of allowing cancellation?).